packages/airnode-node/src/handlers/process-http-signed-data-request.ts

```diff
@@ -17,29 +19,43 @@ export async function processHttpSignedDataRequest(
 
   const endpoints = find(config.ois, ['title', trigger.oisTitle])?.endpoints;
   const endpoint = find(endpoints, ['name', trigger.endpointName]);
+  const decodedParameters = evm.encoding.safeDecode(parameters);
+
   if (!endpoint) {
     return [new Error(`No endpoint definition for endpoint ID '${endpointId}'`), null];
   }
 
   // TODO: There should be an TS interface for required params
-  if (!parameters._templateId) {
-    return [
-      new Error(`You must specify "_templateId" for the requestId/subscriptionId in the request parameters.`),
-      null,
-    ];
+  if (!decodedParameters) {
+    return [new Error(`Request contains invalid encodedParameters: ${parameters}`), null];
   }
 
   const requestId = randomHexString(16);
   const logOptions = buildBaseOptions(config, { requestId });
   const airnodeAddress = wallet.getAirnodeWallet(config).address;
+
+  const template = <ApiCallTemplate>{
+    airnodeAddress,
+    endpointId,
+    encodedParameters: parameters,
+  };
+  const templateId = getExpectedTemplateId(template);
+
   const aggregatedApiCall: AggregatedApiCall = {
     type: 'http-signed-data-gateway',
     id: requestId,
     airnodeAddress,
     endpointId,
     endpointName: trigger.endpointName,
     oisTitle: trigger.oisTitle,
-    parameters,
+    parameters: decodedParameters,
+    templateId: templateId,
+    template: {
+      airnodeAddress,
+      endpointId,
```

Comment:
It's weird to me to have some of these fields duplicated. Is this necessary?

Reply:
I guess it depends on where do you want to construct the apiCall object. You can do it inside `callApi` function but I like this better even with some duplicity.

Reply:
I think this duplication happens in the regular `apiCalls` as well so it should be fine to do this since these fields are used in the `callApi` methods (verifyTemplate)